### PR TITLE
Ato 626 porting

### PR DESCRIPTION
**Pull request by sanchariGr**
**Proposed changes**:
- Moving the commits from PR https://github.com/RasaHQ/rasa/pull/11902/commits into 3.4.x.

### CODE REVIEW

File: .github/workflows/continous-integration.yml
Comment (ancalita) [inline .github/workflows/continous-integration.yml:1049]: Missing new line.

File: rasa/core/actions/action.py
Comment (ancalita) [inline rasa/core/actions/action.py:741]: This var needs the type hint back `Any` so that the code quality mypy check passes.

**Review by ancalita — APPROVED** [review]
Two minor details need to be sorted out before merging 👍🏻

**Comment by ancalita** [comment]
@sanchariGr I would hold off from merging this PR until Monday because of this [thread](https://rasa-hq.slack.com/archives/C01HHMR4X8S/p1674650314996349).

**Comment by sanchariGr** [comment]
> @sanchariGr I would hold off from merging this PR until Monday because of this [thread](https://rasa-hq.slack.com/archives/C01HHMR4X8S/p1674650314996349).

Ah thanks for pointing me to the thread @ancalita , i'll hold off.

**Review by radovanZRasa — COMMENTED** [review]
Postponing this PR until we resolve discussion on https://github.com/RasaHQ/rasa/pull/11953#pullrequestreview-1277079251

**Pull request closed without merging**
